### PR TITLE
Feature/pending message

### PR DESCRIPTION
**Pull request by AdityaKhatri**

### CODE REVIEW

File: src/components/Cover/index.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    const resizeObserverRef = React.useRef<ResizeObserver>();
+    const mutationObserverRef = React.useRef<MutationObserver>();
+    const callbackRef = React.useRef<number | undefined>();
+
+    const queueSync = React.useCallback(() => {
+        if (callbackRef.current) {
+            window.cancelIdleCallback(callbackRef.current);
+        }
+
+        callbackRef.current = window.requestIdleCallback(() => {
```
Comment (tnagorra) [inline src/components/Cover/index.tsx:31]: Looks like this doesn't work on safari at all!

Comment (tnagorra) [reply src/components/Cover/index.tsx:31]: https://caniuse.com/requestidlecallback

Comment (AdityaKhatri) [reply src/components/Cover/index.tsx:31]: I've added the polyfills

File: src/components/Cover/index.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                containerRef.current.style.top = `${top}px`;
+                containerRef.current.style.width = `${width}px`;
+                containerRef.current.style.height = `${height}px`;
+            }
+        }, { timeout: 200 });
+    }, []);
+
+    const handleResize = React.useCallback(() => {
+        queueSync();
+    }, [queueSync]);
```
Comment (tnagorra) [inline src/components/Cover/index.tsx]: Any reason to implement this function?

File: src/components/Cover/index.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    const handleAttributeMutation = React.useCallback((e) => {
+        if (e[0].target === containerRef.current) {
+            queueSync();
+        }
+    }, [queueSync]);
+
+    const handleScroll = React.useCallback(() => {
+        queueSync();
+    }, [queueSync]);
```
Comment (tnagorra) [inline src/components/Cover/index.tsx]: Any reason to implement this function?

File: src/components/Cover/index.tsx (hunk 1 above)
Comment (tnagorra) [reply src/components/Cover/index.tsx:31]: Have you imported the poly-fill and set it as side-effect? Maybe we can add this as peer-dependency?

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -39,12 +39,14 @@
...
         "@togglecorp/fujs": "^1.9.2-beta.0",
+        "@types/requestidlecallback": "^0.3.1",
         "d3-axis": "^2.0.0",
         "d3-scale": "^3.2.3",
         "file-saver": "^2.0.2",
         "react-icons": "^3.11.0",
         "react-textarea-autosize": "^8.3.3",
-        "resize-observer": "^1.0.0"
+        "resize-observer": "^1.0.0",
+        "resize-observer-polyfill": "^1.5.1"
```
Comment (tnagorra) [inline package.json]: Why do we need both of them?

**Pull request merged**